rootex/framework/components/visual/model/renderable_component.h

```diff
@@ -18,7 +18,7 @@ class RenderableComponent : public Component
 	float m_LODBias;
 	float m_LODDistance;
 
-	HashMap<Ref<MaterialResourceFile>, Ref<MaterialResourceFile>> m_MaterialOverrides;
+	HashMap<MaterialResourceFile*, Ref<MaterialResourceFile>> m_MaterialOverrides;
```

Comment:
Why does it not want to own the material overrides? If the previous material was destroyed, you would immediately crash when you try to revert a material override.

Reply:
@sin3point14

Reply:
`m_MaterialOverrides` is just acting like a dictionary from which you can ask for what overrides this material. It shouldn't be owning that material. the rightful owner of the material extracted from a model file is ModelResourceFile https://github.com/sdslabs/Rootex/blob/e6af1008be794b922510bb560812551c87ab27b0/rootex/core/resource_files/model_resource_file.h#L14
Since it will be guaranteed that when this hashmap is accessed it will be through a material retrieved from `ModelResourceFile` so we won't be having any crashes. Also the access logic remains exactly the same since https://en.cppreference.com/w/cpp/memory/shared_ptr/hash hashes a shared_ptr from `get()`
We were also facing some weird issues in the Lua API due to `shared_ptr`(which i forgot, will get back with that once i am able to replicate that) and I think Lua scripts shouldn't have ownership of anything unless they created it.